### DISCUSSION TITLE
approve-multitenant-permission-scope-spfx-solution/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#620] [Comments]
# approve-multitenant-permission-scope-spfx-solution/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/approve-multitenant-permission-scope-spfx-solution/

**Comment by Ms-fernandes** [comment]
Hi, I am tring to connect to aad secured web api through this method, but i get error saying admin has not consented using of this app.

**Reply by estruyf** [reply]
There is no service principal if the admin hasn't consented to the app yet. Have you tried the PowerShell, bash, or consent URL?

**Comment by AdamBless14** [comment]
I am using the correct resource names noted here:  

"webApiPermissionRequests": [
      {
        "resource": "spfxaadclient",
        "scope": "user_impersonation"
      },
      {
        "resource": "apiaadclient",
        "scope": "access_as_user"
      }
    ]

I also have granted admin consent on azure when configuring the permissions but still get this error

**Reply by estruyf** [reply]
You can always try manually adding permission on the `SharePoint Online Web Client Extensibility` Entra app on Azure. If that would fail as well, there might be something wrong with the resource names.
